simpledotorg/simple-server
# Update materialised summary to capture more fields for patient line list

**Story card:** [sc-8205](https://app.shortcut.com/simpledotorg/story/8205/generate-line-lists-using-a-materialized-view)

## Because

Generating line lists for large regions is very slow and resource intensive. We have received complaints about the line list download taking upwards of 30 minutes, and have also observed our sidekiq boxes run out of resources when multiple line list download jobs are triggered (often by the same user waiting for their earlier downloads to come through). 

## This addresses

This PR converts our line list into a materialised Postgres view that gets updates on a daily basis. This precomputes and materialises the patient summary for all the patients, so we don't have to run additional queries and load unnecessary records in memory. 

## Test instructions

Download the patient line list from the district reports page while observing the active queries in postgres. It should not make additional queries to table other than materialized_patient_summaries.

```
\watch 1 "SELECT procpid, age(clock_timestamp(), query_start), usename, current_query FROM pg_stat_activity WHERE current_query != '<IDLE>' AND current_query NOT ILIKE '%pg_stat_activity%'  ORDER BY query_start desc";
```
